src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperContextExtensions.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Logging;
+using Microsoft.AspNet.Mvc.TagHelpers;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Utility related extensions for <see cref="TagHelperContext"/>.
+    /// </summary>
+    public static class TagHelperContextExtensions
+    {
+        /// <summary>
+        /// Determines whether a <see cref="TagHelper" /> should run based on the presence of attributes it requires.
+        /// </summary>
+        /// <param name="context">The <see cref="TagHelperContext"/>.</param>
+        /// <param name="requiredAttributes">The attributes the <see cref="TagHelper" /> requires in order to run.</param>
+        /// <param name="logger">An optional <see cref="ILogger"/> to log warning details to.</param>
+        /// <returns>A <see cref="bool"/> indicating whether the <see cref="TagHelper" /> should run.</returns> 
+        public static bool ShouldProcess(this TagHelperContext context, IEnumerable<string> requiredAttributes, ILogger logger = null)
+        {
+            if (context == null)
+            {
+	           throw new ArgumentNullException("context");
+            }
+
+            if (requiredAttributes == null)
+            {
+	           throw new ArgumentNullException("requiredAttributes");
+            }
+			
+            // Check for all attribute values & log a warning if any required are missing
+            var atLeastOnePresent = false;
+            var missingAttrNames = new List<string>();
+
+            foreach (var attr in requiredAttributes)
+            {
+                if (!context.AllAttributes.ContainsKey(attr)
+                    || context.AllAttributes[attr] == null
```

Comment:
It feels like the act of being not `null` and not `empty` meaning "not provided" is something that is very `TagHelper` specific; some may be 100% ok with a whitespace or `null` value, as long as it was provided by the user. 

Ultimately this method is a utility method that any `TagHelper` author can use so we should make sure it doesn't restrict itself or confuse users.  I'd rather do less in this case than do more; people are more likely to write `<something foo="@myVar" />` where `@myVar` is whitespace or `null` (would cause lots of confusion as to why the `TagHelper` wouldn't be running because the attribute was provided).

If you absolutely want to keep the functionality of treating `null`/whitespace as not being provided add a `<remark>` to define what "required" means to this method.